pandas/tests/frame/test_alter_axes.py

```diff
@@ -1186,6 +1187,45 @@ def test_reset_index_multiindex_nan(self):
         rs = df.set_index(["A", "B"]).reset_index()
         tm.assert_frame_equal(rs, df)
 
+    def test_reset_index_multiindex_datetime_all_nan(self):
+        # GH 19602
+        df = DataFrame({0: DatetimeIndex([]), 1: []}, columns=[0, 1])
+        rs = df.set_index([0, 1]).reset_index()
+        tm.assert_frame_equal(rs, df)
+
```

Comment:
can you try to parameterize these rather than copy-pasting (you can include existing tests if need be)